src/main/java/de/tum/cit/aet/artemis/communication/web/FaqResource.java

```diff
@@ -210,7 +200,39 @@ public ResponseEntity<Set<FaqDTO>> getAllFaqsForCourseByStatus(@PathVariable Lon
     public ResponseEntity<Set<String>> getFaqCategoriesForCourse(@PathVariable Long courseId) {
         log.debug("REST request to get all Faq Categories for the course with id : {}", courseId);
         Set<String> faqs = faqRepository.findAllCategoriesByCourseId(courseId);
+        return ResponseEntity.ok().body(faqs);
+    }
 
+    @GetMapping("courses/{courseId}/faq-categories/{faqState}")
+    @EnforceAtLeastStudentInCourse
+    public ResponseEntity<Set<String>> getFaqCategoriesForCourseByState(@PathVariable Long courseId, @PathVariable FaqState faqState) {
+        log.debug("REST request to get all Faq Categories for the course with id : {} and the state {}", courseId, faqState);
+        checkShouldAccessNotAccepted(faqState, courseId);
+        Set<String> faqs = faqRepository.findAllCategoriesByCourseIdAndState(courseId, faqState);
         return ResponseEntity.ok().body(faqs);
     }
+
+    /**
+     * @param courseId the id of the course the faq belongs to
+     * @param role     the required role of the user
+     * @throws AccessForbiddenException if the user does not have at least role
+     *
+     */
+    private void checkRoleForCourse(Long courseId, Role role) {
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(role, course, null);
+    }
+
+    private void checkShouldAccessNotAccepted(FaqState faqState, Long courseId) {
+        if (faqState != FaqState.ACCEPTED) {
+            checkRoleForCourse(courseId, Role.TEACHING_ASSISTANT);
+        }
+    }
+
+    private void checkIsInstructor(FaqState faqState, Long courseId) {
```

Comment:
```suggestion
    private void checkIsInstructorForAcceptedFaq(FaqState faqState, Long courseId) {
```
Otherwise I would expect a normal check if the user is an instructor